packages/apollo-gateway/src/__tests__/gateway/lifecycle-hooks.test.ts

```diff
@@ -29,6 +30,22 @@ const serviceDefinitions = [
   url: `http://localhost:${i}`,
 }));
 
+let logger: Logger;
+
+beforeEach(() => {
+  const warn = jest.fn();
+  const debug = jest.fn();
+  const error = jest.fn();
+  const info = jest.fn();
+
+  logger = {
+    warn,
+    debug,
+    error,
+    info,
+  };
+});
+
```

Comment:
I have also been repeating this pattern in many places.  Perhaps at some point soon (not now), we should just make a `spyableLogger`?